python/common/org/python/types/List.java

```diff
@@ -337,14 +337,22 @@ public boolean __setattr_null(java.lang.String name, org.python.Object value) {
                     if (step < 0) {
                         long start;
                         if (slice.start != null) {
-                            start = Math.min(slice.start.value, this.value.size());
+                            if (slice.start.value < 0) {
+                                start = this.value.size() + slice.start.value;
```

Comment:
Does this take account if `start` is less than 0? If not it should have a `Math.max` around it as well (similar to the positive case).